hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -249,6 +249,7 @@ private static DataStream<HoodieRecord> streamBootstrap(
 
     if (conf.getBoolean(FlinkOptions.INDEX_BOOTSTRAP_ENABLED) || bounded) {
       dataStream1 = dataStream1
+          .keyBy(HoodieRecord::getRecordKey)
           .transform(
```

Comment:
Why shuffle the input by primary keys here ? We already shuffle the output of bootstrap operators by primary keys.

Reply:
![image](https://user-images.githubusercontent.com/36881649/174066220-4e96fa32-e2de-4f13-a30f-6b9ff2341640.png)
The sequence of data received by the bucketAssign may change.